### PR TITLE
 Add a file deletion service to remove input/output files after their time to live expires

### PR DESCRIPTION
**Pull request by daniel-warren**
Change adds a new generic FileDeletionService class that will scan provided directories for files over a specified age, and then deletes them. 

Then uses this service in a basic configuration to run for the input/output folders, making use of the new properties file feature to configure TTL, enable/disable the deletion & the frequency that it runs at. 

This PR is current destination is the properties branch (only the changes relevant with this branch are shown in the file comparison) however, once this change is merged into develop, this base/target should be updated to be develop instead.

### CODE REVIEW

**Comment by daniel-warren** [comment]
It would seem that the file comparison still includes all previous changes that are present in the develop branch & in feature/microservice_properties_file. Updated base branch accordingly. I suggest looking at https://github.com/idrsolutions/base-microservice-example/pull/49/commits/adc20c6c7aa1c41748a27c53b3d3dbeee10a4bed to see the changes made for this PR.

File: src/main/java/com/idrsolutions/microservice/utils/FileDeletionService.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+public class FileDeletionService {
+    private static final Logger LOG = Logger.getLogger(FileDeletionService.class.getName());
+
+    private final ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
+
+    /**
+     * Create an instance of the FileDeletionService
+     *
+     * @param dirs the root directories that will be scanned for files that should be deleted
+     * @param fileLifeSpan the life span of the files before they should be deleted
```
Comment (JacobIDR) [inline src/main/java/com/idrsolutions/microservice/utils/FileDeletionService.java:34]: Probably worth specifying the units for the lifespan too

File: src/main/java/com/idrsolutions/microservice/utils/FileDeletionService.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        setUpService(dirs, fileLifeSpan, frequency);
+    }
+
+    /**
+     * Use a ScheduledExecutorService to set up a service with the process of deleting any files that exceed the fileLifeSpan
+     *
+     * This method does not remove any previously set scheduled tasks from the ExecutorService
+     *
+     * @param dirs the root directories that will be scanned for files that should be deleted
+     * @param fileLifeSpan the life span of the files before they should be deleted
```
Comment (JacobIDR) [inline src/main/java/com/idrsolutions/microservice/utils/FileDeletionService.java:47]: Probably worth specifying the units for the lifespan too

**Review by JacobIDR — CHANGES_REQUESTED** [review]

File: src/main/java/com/idrsolutions/microservice/utils/FileDeletionService.java (hunk 2 above)
Comment (daniel-warren) [reply src/main/java/com/idrsolutions/microservice/utils/FileDeletionService.java]: @JacobIDR Thank you for spotting. I have resolved in the latest commit

**Review by JacobIDR — APPROVED** [review]

**Review by KieranFrance — APPROVED** [review]

**Review by leonatherton — APPROVED** [review]

**Pull request merged**